### ISSUE TITLE
Version 3.3.1 GridView throwing undefined offset in Line 1227

### ISSUE THREAD

**Issue by SSiwek** [kartik-v/yii2-grid#907]
Hi,

I just run into that Error. I fixed it with:

public function renderPageSummary()
[...]
        //for ($i = 0; $i < $cols; $i++) {
        foreach($this->columns as $i => $column){
            /** @var DataColumn $column */
            //$column = $this->columns[$i];
            if (!method_exists($column, 'renderPageSummaryCell')) {
                $cells[] = Html::tag('td');
                continue;
            }
            $cells[] = $column->renderPageSummaryCell();

Sven

**Comment by execut** [comment]
Minimal code to reproduce the bug:
```php
$dataProvider = new \yii\data\ArrayDataProvider([
    'allModels' => [
        ['id' => 1]
    ],
]);
echo \kartik\grid\GridView::widget([
    'dataProvider' => $dataProvider,
    'showPageSummary' => true,
    'columns' => [
        'id' => 'id',
    ]
]);
```

**Comment by kartik-v** [comment]
Fixed via #908.
